### PR TITLE
Changed readme

### PR DESCRIPTION
**Pull request by yifeiwu**
I added my demo app to this and a few other small edits to the documentation. I hope it's not too obtrusive.

### CODE REVIEW

**Comment by onebree** [comment]
@yifeiwu I would only suggest you squash your commits into 1. Since you are updating 1 file, it will result in a cleaner commit history for this repo

File: README.md [hunk 1]
```diff
@@ -51,45 +51,36 @@ Install it
...
-### Rails 4.x.x
+Add to your `app/assets/javascripts/application.js` file
 
-Run the generator
-
-    rails generate autocomplete:install
-
-And include autocomplete-rails.js on your layouts
-
-    javascript_include_tag "autocomplete-rails.js"
```
Comment (onebree) [inline README.md]: Why did you remove this suggestion? I added it to my application.html.haml and it worked.

File: README.md [hunk 2]
```diff
@@ -51,45 +51,36 @@ Install it
...
 
-## Usage
+3. The autocomplete is implemented in the context of a search form with a simple scope search(see the food model and controller). 
 
 ### Model Example
 
-Assuming you have a Brand model:
+Assuming you have a Brand model that contains a name attribute:
 
     class Brand < ActiveRecord::Base
```
Comment (onebree) [inline README.md]: Rather than the class, I suggest entering a sample schema table, that shows the attributes and their data types.

File: README.md (hunk 1 above)
Comment (yifeiwu) [inline README.md]: I believe these instructions are remnants from the Rails 3.x.x instructions. For Rails 4 which this gem is designed for, you only need what I've left in.

Comment (onebree) [inline README.md]: Technically, this gem is for Rails 3 AND 4, as it is the currently maintained repo. I think it is best to condense any configuration into 1 section (not by Rails Version), and just add a note: `When updating the gem, it is best to regenerate the javascript file, to ensure all fixes are made` or something like that

**Comment by bigtunacan** [comment]
@yifeiwu as @onebree noted this gem supports multiple versions of Rails.  This was the reason for the name change from `rails3-jquery-autocomplete` to `rails-jquery-autocomplete` to help make it clearer that this is one gem intended for current and future versions of Rails.  The references in the README to instructions related to Rails 3 should not be removed until the gem no longer supports Rails 3.  At such a time that I decide (out of necessity) to no longer support feature changes for Rails 3 then I would created a tag or possible a permanent branch where people can still see the documentation for the Rails 3 version that may still need it.

**Comment by bigtunacan** [comment]
@yifeiwu To add to that though; I did indicate in the README.md to continue using the old gem for Rails 3. Originally I was planning to break compatibility with Rails 3 earlier, but have since decided it isn't necessary yet.  I will update the README to better reflect this.

**Comment by yifeiwu** [comment]
Fair enough, thanks for the clarification. I haven't tested this on Rails 3 and I didn't know if people would be better off using the older branch.

**Comment by onebree** [comment]
@yifeiwu As long as the Rails 3 branch is kept (relatively) up-to-date, then a separate branch is good. You can define a certain branch in your Gemfile. That way, master will always relate to the latest support rails version.

The Rails project actually has a branch for every stable release. It allows you to deploy 2 versions of something, rather than 1 one-size-fits-all (with errors) gem, or 2 gems in separate repos each.

File: README.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ Install it
 
     bundle install
 
-### Rails 4.x.x
+### Rails < 4.x.x
```
Comment (onebree) [inline README.md]: remove this! It is saying Rails older than 4.x.x
